fix_checkpoint_history.py

```
@@ -0,0 +1,80 @@
+    if history_file.exists():
+        with open(history_file, 'r') as f:
+            history = json.load(f)
+    
+    # Add entry for this file
+    if target_csv not in history:
+        history[target_csv] = []
+    
+    history[target_csv].append({
+        "checkpoint_file": str(checkpoint_file),
+        "location_index": estimated_locations,
+        "timestamp": datetime.now().isoformat(),
```

Ruff (0.12.2)

`datetime.datetime.now()` called without a `tz` argument

(DTZ005)